README.md

```diff
@@ -39,54 +39,40 @@ ant test
-
-<sup>1</sup>Replace the `yahoo` part of the key name with the name of the client (e.g. `yahoo`, `google`, `outlook`).
+This service's configuration for OAuth clients are setup in Ldap attributes on a global or domain level.
 
+**Ldap Properties**
 
-**Localconfig Client Specific Properties**
+See the [client setup wiki].
 
-**Yahoo Implementation Properties**
+| Key | Description | Required By | Template |
+| --- | ----------- | ----------- | -------- |
+| zimbraOAuthConsumerCredentials | OAuth credentials for a client, set at global config or domain level. | All | `<client-id>:<client-secret>:<client>` |
+| zimbraOAuthConsumerRedirectUri | The callback where the client returns the user too. | All | `http[s]://<domain[:port]>/service/extension/oauth2/authenticate/<client>:<client>` |
+| zimbraOAuthConsumerAPIScope | The scopes required to access user data. | Google, Facebook | `<scope1>+<scope2>+...:<client>` |
 
-| Key | Description | Optional | Example Options |
-| --- | ----------- | -------- | --------------- |
-| zm_oauth_yahoo_client_id | The Yahoo app's client id | | |
-| zm_oauth_yahoo_client_secret | The Yahoo app's client secret | | |
-| zm_oauth_yahoo_client_redirect_uri | The callback Yahoo returns the user to | | `https://this.service.host.com/service/extension/oauth2/authenticate/yahoo` |
-
-
-**Google Implementation Properties**
+***Client specific scopes to use with the zimbraOAuthConsumerAPIScope config***
 
-| Key | Description | Optional | Example Options |
-| --- | ----------- | -------- | --------------- |
-| zm_oauth_google_client_id | The Google app's client id | | |
-| zm_oauth_google_client_secret | The Google app's client secret | | |
-| zm_oauth_google_client_redirect_uri | The callback Google returns the user to | | `https://this.service.host.com/service/extension/oauth2/authenticate/google` |
-| zm_oauth_google_scope | The token scope to request | | `profile` |
+| Client | Required scopes string |
+| ------ | ---------------------- |
+| Google | `https://www.googleapis.com/auth/contacts profile:google` |
+| Facebook | `user_friends,read_custom_friendlists,email,user_location,public_profile,user_about_me,user_birthday,groups_access_member_info:facebook` |
 
+Note: Delimiters can vary across clients.
 
-**Facebook Implementation Properties**
+**Localconfig General Properties**
 
 | Key | Description | Optional | Example Options |
 | --- | ----------- | -------- | --------------- |
-| zm_oauth_facebook_client_id | The Facebook app's client id | | |
-| zm_oauth_facebook_client_secret | The Facebook app's client secret | | |
-| zm_oauth_facebook_client_redirect_uri | The callback Facebook returns the user to | | `https://this.service.host.com/service/extension/oauth2/authenticate/facebook` |
-| zm_oauth_facebook_scope | The token scope to request | | `profile` |
+| host_uri_template | The host uri to connect via ZMailbox | Yes | `https://%s:443` |
```

Comment:
The `zm_oauth_classes_handlers_yahoo<sup>1</sup>` is still needed for non-known keys.